### ISSUE TITLE
if settings() present and pixelDensity() is in setup(), nothing set/no error

### ISSUE THREAD

**Issue by neilpanchal** [processing/processing#4703]
Found this a bit odd and I am not sure if this is a bug or an known issue / expected behavior.

when using `pixelDensity()` inside `settings()` as:

``` processing
void settings() {
    size(1024, 1024, P2D);
    pixelDensity(2);
}
```

and calling `saveFrame()` in the `draw()` loop, the saved image size **2048x2048** pixels.

However, when using `pixelDensity()` inside `setup()` as:

``` processing
void settings() {
    size(1024, 1024, P2D);
}

void setup() {
    pixelDensity(2);
}
```

and calling `saveFrame()` in the `draw()` loop, the saved image size is **1024x1024** pixels.

Not sure if this is an odd behavior, just wanted to open an issue and report it.

I prefer saving 2x resolution frames. 

Thanks,
Neil

**Comment by benfry** [comment]
You can only use `pixelDensity()` inside `setup()`: https://processing.org/reference/pixelDensity_.html

Did it not give you an error message in the console?

**Comment by benfry** [comment]
Confirmed that it doesn't give any sort of error, so I've changed the bug title to represent what's going on. It should be repeating what I wrote in the last comment (what's in the reference).

**Comment by benfry** [comment]
Fixed for 3.5.
